### PR TITLE
Ozonosfer test configuration

### PR DESCRIPTION
**Pull request by EnverKESMEN**
* all tests were moved to the test package

* whitelist added to redis for rate limiting

* ozonosfer is now more lightweight

Co-authored-by: Nur Erkartal <nur.erkartal@iyzico.com>

### CODE REVIEW

File: ozonosfer-spring/src/main/java/com/iyzico/ozonosfer/domain/service/ListService.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package com.iyzico.ozonosfer.domain.service;
+
+public interface ListService {
```
Comment (yyenigun) [inline ozonosfer-spring/src/main/java/com/iyzico/ozonosfer/domain/service/ListService.java]: ListService name is a bit generic. RateLimitTogglingService can be a better naming. For example: RateLimitTogglingService.isRateLimitEnabled(String key), isRateLimitDisabled(String key) is more readable.

File: ozonosfer-spring/src/main/java/com/iyzico/ozonosfer/infrastructure/service/RedisBasicRateLimiterService.java [hunk 2]
```diff
@@ -39,9 +39,12 @@
     private static final double DELTA = 1.0;
 
     private RedisTemplate<String, String> redisTemplate;
+    private RedisWhiteListService redisWhiteListService;
 
-    public RedisBasicRateLimiterService(RedisTemplate<String, String> redisTemplate) {
+    public RedisBasicRateLimiterService(RedisTemplate<String, String> redisTemplate,
+                                        RedisWhiteListService redisWhiteListService) {
         this.redisTemplate = redisTemplate;
+        this.redisWhiteListService = redisWhiteListService;
```
Comment (yyenigun) [inline ozonosfer-spring/src/main/java/com/iyzico/ozonosfer/infrastructure/service/RedisBasicRateLimiterService.java]: RedisRateLimitTogglingService is better naming.

File: ozonosfer-spring/src/main/java/com/iyzico/ozonosfer/infrastructure/service/RedisBasicRateLimiterService.java [hunk 3]
```diff
@@ -63,14 +66,18 @@ public RedisBasicRateLimiterService(RedisTemplate<String, String> redisTemplate)
                     name = "coreSize",
                     value = "20")})
     public void rateLimit(RateLimitRequest request) {
-        Long count = retrieveCount(request);
-        if (rateLimitExceeded(request.getLimit(), count)) {
-            throw new RateLimitedException(request);
-        } else {
-            incrementCount(request);
+        if (redisWhiteListService.isNotLimited(String.valueOf(request.getKey()))) {
```
Comment (yyenigun) [inline ozonosfer-spring/src/main/java/com/iyzico/ozonosfer/infrastructure/service/RedisBasicRateLimiterService.java]: I couldn't understand redisWhiteListService.isNotLimited well. if(rateLimitTogglingService.isRateLimitEnabled(key)) will also be better here. if(!true) is not as readable as if(true).

File: ozonosfer-spring/src/main/java/com/iyzico/ozonosfer/infrastructure/service/RedisWhiteListService.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
+package com.iyzico.ozonosfer.infrastructure.service;
+
+import com.iyzico.ozonosfer.domain.service.ListService;
+import org.springframework.data.redis.core.RedisTemplate;
+import org.springframework.stereotype.Service;
+
+@Service
+public class RedisWhiteListService implements ListService {
+
+    private static final String OZONOSFER_WHITE_LIST = "ozonosfer-white-list";
```
Comment (yyenigun) [inline ozonosfer-spring/src/main/java/com/iyzico/ozonosfer/infrastructure/service/RedisWhiteListService.java]: ozon-enabled-list can be better key for redis.

File: ozonosfer-spring/src/main/java/com/iyzico/ozonosfer/domain/service/ListService.java (hunk 1 above)
Comment (EnverKESMEN) [reply ozonosfer-spring/src/main/java/com/iyzico/ozonosfer/domain/service/ListService.java]: I was not sure when choosing this name. Suggestion looks better. I'm changing.

Comment (EnverKESMEN) [reply ozonosfer-spring/src/main/java/com/iyzico/ozonosfer/domain/service/ListService.java]: I was not sure when choosing this name. Suggestion looks better. I'm changing the naming.

**Pull request merged**